File: leetcode-daily/01-april-2025/2140.go
func mostPoints(questions [][]int) int64 {
    n := len(questions)
    dp := make([]int64, n)
    for i := range dp {
        dp[i] = -1
    }
    dp[n - 1] = int64(questions[n - 1][0])

    for i := n - 2; i >= 0; i-- {
        nextIndex := i + questions[i][1] + 1
        solve := int64(questions[i][0]) 
        if nextIndex < n {
            solve += dp[nextIndex]
        }
        dp[i] = max(solve, dp[i + 1])
    }

    return dp[0]
}

File: leetcode-daily/01-april-2025/2140.java
class Solution {
    public long mostPoints(int[][] questions) {
        int n = questions.length;
        long[] dp = new long[n];
        dp[n - 1] = questions[n - 1][0];

        for (int i = n - 2; i >= 0; i--) {
            int nextIndex = i + questions[i][1] + 1;
            long solve = questions[i][0] + (nextIndex < n ? dp[nextIndex] : 0);
            dp[i] = Math.max(solve, dp[i + 1]);
        }

        return dp[0];
    }
}